Unrecognized arguments local-rank in multi-gpu train in self_supervised_pretraining

```
root@6091477:/opt/toolkit/tutorials/monai/self_supervised_pretraining/vit_unetr_ssl/multi_gpu# python -m torch.distributed.launch --nproc_per_node=$(nvidia-smi -L | wc -l) mgpu_ssl_train.py --batch_size=8 --epochs=10 --base_lr=2e-4 --logdir_path=/var/log/ssl_train --json_path=../datalists/tcia/dataset_split.json
/usr/local/lib/python3.10/dist-packages/torch/distributed/launch.py:183: FutureWarning: The module torch.distributed.launch is deprecated
and will be removed in future. Use torchrun.
Note that --use-env is set by default in torchrun.
If your script expects `--local-rank` argument to be set, please
change it to read from `os.environ['LOCAL_RANK']` instead. See 
https://pytorch.org/docs/stable/distributed.html#launch-utility for 
further instructions

  warnings.warn(
[2024-03-15 09:16:37,155] torch.distributed.run: [WARNING] 
[2024-03-15 09:16:37,155] torch.distributed.run: [WARNING] *****************************************
[2024-03-15 09:16:37,155] torch.distributed.run: [WARNING] Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
[2024-03-15 09:16:37,155] torch.distributed.run: [WARNING] *****************************************
usage: ViT Self-Supervised Learning [--data_root DATA_ROOT] [--json_path JSON_PATH] [--logdir_path LOGDIR_PATH] [--output PATH] [--local_rank LOCAL_RANK] [--epochs EPOCHS] [--batch_size BATCH_SIZE] [--base_lr BASE_LR] [--seed SEED] [--deterministic]
ViT Self-Supervised Learning: error: unrecognized arguments: --local-rank=1
usage: ViT Self-Supervised Learning [--data_root DATA_ROOT] [--json_path JSON_PATH] [--logdir_path LOGDIR_PATH] [--output PATH] [--local_rank LOCAL_RANK] [--epochs EPOCHS] [--batch_size BATCH_SIZE] [--base_lr BASE_LR] [--seed SEED] [--deterministic]
ViT Self-Supervised Learning: error: unrecognized arguments: --local-rank=0
[2024-03-15 09:16:52,174] torch.distributed.elastic.multiprocessing.api: [ERROR] failed (exitcode: 2) local_rank: 0 (pid: 2009) of binary: /usr/bin/python
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.10/dist-packages/torch/distributed/launch.py", line 198, in <module>
    main()
  File "/usr/local/lib/python3.10/dist-packages/torch/distributed/launch.py", line 194, in main
    launch(args)
  File "/usr/local/lib/python3.10/dist-packages/torch/distributed/launch.py", line 179, in launch
    run(args)
  File "/usr/local/lib/python3.10/dist-packages/torch/distributed/run.py", line 825, in run
    elastic_launch(
  File "/usr/local/lib/python3.10/dist-packages/torch/distributed/launcher/api.py", line 137, in __call__
    return launch_agent(self._config, self._entrypoint, list(args))
  File "/usr/local/lib/python3.10/dist-packages/torch/distributed/launcher/api.py", line 271, in launch_agent
    raise ChildFailedError(
torch.distributed.elastic.multiprocessing.errors.ChildFailedError: 
============================================================
mgpu_ssl_train.py FAILED
------------------------------------------------------------
Failures:
[1]:
  time      : 2024-03-15_09:16:52
  host      : 6091477
  rank      : 1 (local_rank: 1)
  exitcode  : 2 (pid: 2010)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
------------------------------------------------------------
Root Cause (first observed failure):
[0]:
  time      : 2024-03-15_09:16:52
  host      : 6091477
  rank      : 0 (local_rank: 0)
  exitcode  : 2 (pid: 2009)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
============================================================
```

--- a/self_supervised_pretraining/vit_unetr_ssl/README.md
+++ b/self_supervised_pretraining/vit_unetr_ssl/README.md
@@ -144,7 +144,7 @@ At the time of creation of this tutorial, the below additional dependencies are 
 To begin training with 2 GPU's please see the below example command for execution of the SSL multi-gpu training
 script:
 
-`CUDA_VISIBLE_DEVICES=0,1 python -m torch.distributed.launch --nproc_per_node=2 mgpu_ssl_train.py --batch_size=8 --epochs=500 --base_lr=2e-4 --logdir_path=/to/be/defined --output=/to/be/defined --data_root=/to/be/defined --json_path=/to/be/defined`
+`CUDA_VISIBLE_DEVICES=0,1 torchrun --nproc_per_node=2 mgpu_ssl_train.py --batch_size=8 --epochs=500 --base_lr=2e-4 --logdir_path=/to/be/defined --output=/to/be/defined --data_root=/to/be/defined --json_path=/to/be/defined`
 
 It can be configured to launch on more GPU's by adding the relevant `CUDA Device` ID in `CUDA_VISIBLE_DEVICES`
 and increasing the total count of GPU's `--nproc_per_node`

--- a/self_supervised_pretraining/vit_unetr_ssl/multi_gpu/mgpu_ssl_train.py
+++ b/self_supervised_pretraining/vit_unetr_ssl/multi_gpu/mgpu_ssl_train.py
@@ -137,7 +137,6 @@ def main(args):
         data_list_file_path=json_path, is_segmentation=False, data_list_key="validation", base_dir=data_root
     )
 
-
     print("Total training data are {} and validation data are {}".format(len(train_list), len(val_list)))
 
     train_dataset = CacheDataset(data=train_list, transform=train_transforms, cache_rate=1.0, num_workers=4)
@@ -186,7 +185,7 @@ def main(args):
     optimizer = torch.optim.Adam(model.parameters(), lr=args.base_lr)
 
     model = torch.nn.parallel.DistributedDataParallel(
-        model, device_ids=[int(os.environ['LOCAL_RANK'])], broadcast_buffers=False, find_unused_parameters=True
+        model, device_ids=[int(os.environ["LOCAL_RANK"])], broadcast_buffers=False, find_unused_parameters=True
     )
     model_without_ddp = model.module
 

--- a/self_supervised_pretraining/vit_unetr_ssl/multi_gpu/mgpu_ssl_train.py
+++ b/self_supervised_pretraining/vit_unetr_ssl/multi_gpu/mgpu_ssl_train.py
@@ -67,8 +67,6 @@ def parse_option():
         metavar="PATH",
         help="root of output folder, the full path is <output>/<model_name>/<tag> (default: output)",
     )
-    # Distributed Training
-    parser.add_argument("--local_rank", type=int, help="local rank for DistributedDataParallel")
 
     # DL Training Hyper-parameters
     parser.add_argument("--epochs", default=100, type=int, help="number of epochs")
@@ -139,9 +137,6 @@ def main(args):
         data_list_file_path=json_path, is_segmentation=False, data_list_key="validation", base_dir=data_root
     )
 
-    # TODO Delete the below print statements
-    print("List of training samples: {}".format(train_list))
-    print("List of validation samples: {}".format(val_list))
 
     print("Total training data are {} and validation data are {}".format(len(train_list), len(val_list)))
 
@@ -191,7 +186,7 @@ def main(args):
     optimizer = torch.optim.Adam(model.parameters(), lr=args.base_lr)
 
     model = torch.nn.parallel.DistributedDataParallel(
-        model, device_ids=[args.local_rank], broadcast_buffers=False, find_unused_parameters=True
+        model, device_ids=[int(os.environ['LOCAL_RANK'])], broadcast_buffers=False, find_unused_parameters=True
     )
     model_without_ddp = model.module
 
@@ -340,7 +335,7 @@ if __name__ == "__main__":
     else:
         rank = -1
         world_size = -1
-    torch.cuda.set_device(args.local_rank)
+    torch.cuda.set_device(rank)
     torch.distributed.init_process_group(backend="nccl", init_method="env://", world_size=world_size, rank=rank)
     torch.distributed.barrier()
 

